src/fal/planner/plan.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from typing import Iterator
+
+import networkx as nx
+
+
+def _find_subgraphs(graph: nx.DiGraph) -> Iterator[list[str]]:
+    # Initially topologically sort the graph, and split it
+    # into chunks by using the critical node approach (post-hooks
+    # are also a thing now).
+
+    # Then for each chunk, check whether they have the
+    # same set of ancestors. This is important because
+    # we want to group only the things that are in
+    # the same branch.
+
+    current_stack = []
+    allowed_ancestors = set()
+
+    def split() -> Iterator[list[str]]:
+        if len(current_stack) > 1:
+            yield current_stack.copy()
+        current_stack.clear()
+        allowed_ancestors.clear()
+
+    for node in nx.topological_sort(graph):
+        properties = graph.nodes[node]
+        if properties["kind"] == "python model":
```

Comment:
```suggestion
        if properties["kind"] == NodeKind.FAL_MODEL:
```

There is no `"python model"` kind.